Agents/ForecastingAgent/forecasting/forecasting_agent/agent.py

```diff
@@ -0,0 +1,526 @@
+################################################
+# Authors: Magnus Mueller (mm2692@cam.ac.uk)   #
+# Date: 22 Nov 2022                            #
+################################################
+
+# The purpose of this module is to forecast a time series using a trained model or Prophet
+from dateutil.parser import isoparse
+import datetime as dt
+import os
+import time
+import urllib
+import uuid
+from darts.metrics import mape, mase, mse, rmse, smape
+
+import pandas as pd
+from darts import TimeSeries
+from darts.dataprocessing.transformers import Scaler
+from darts.models import Prophet, TFTModel
+
+from forecasting.datamodel.data_mapping import *
+from forecasting.datamodel.iris import *
+from forecasting.errorhandling.exceptions import KGException
+from forecasting.kgutils.kgclient import KGClient
+from forecasting.kgutils.tsclient import TSClient
+from forecasting.utils.properties import *
+from forecasting.utils.tools import *
+from forecasting.utils.useful_queries import *
+
+#import agentlogging
+# Initialise logger
+#logger = agentlogging.get_logger("prod")
+
+
+def forecast(iri, horizon, forecast_start_date=None, use_model_configuration=None, data_length=None):
+    """
+    Forecast a time series using a pre trained model or Prophet.
+    returns a dictionary with the forecast and some metadata.
+
+    :param iri: The IRI of the time series you want to forecast
+    :param horizon: The number of time steps to forecast
+    :param forecast_start_date: The date from which you want to start forecasting
+    :param use_model_configuration: If you want to force a specific model configuration, you can do so here
+    :param data_length: The number of time steps which should be loaded from the DB
+    :return: The forecast is being returned.
+    """
+    # initialise the  client
+    kgClient = KGClient(QUERY_ENDPOINT, UPDATE_ENDPOINT)
+    tsClient = TSClient(kg_client=kgClient)
+
+    covariates, backtest_series = None, None
+
+    # get the model configuration dictionary from the a specific use_model_configuration or use default
+    if use_model_configuration is not None:
+        model_configuration_name = use_model_configuration
+    else:
+        model_configuration_name = "DEFAULT"
+
+    cfg = MODEL_MAPPING[model_configuration_name].copy()
+    cfg['model_configuration_name'] = model_configuration_name
+    cfg['iri'] = iri
+    cfg['horizon'] = horizon
+    if data_length is not None:
+        cfg['data_length'] = data_length
+        
+    cfg['ts_iri'] = get_ts_iri(cfg, kgClient)
```

Comment:
The way you use `tsIRI` is inconsistent with the way `timeseriesIRI` is actually "_introduced_ in the [TimeSeriesClient](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries#instantiation-in-rdb). You are rather referring to the `dataIRI`. Please resolve that inconsistency to avoid confusion.

```
{
  "fc_model": {
    "train_again": true,
    "name": "prophet",
    "scale_data": false,
    "input_length": 5
  },
  "data_length": 5,
  "model_configuration_name": "DEFAULT",
  "iri": "https://www.theworldavatar.com/kg/pms_dh/HeatDemand_ad132fd9-846f-4b04-9f32-7e57a30cda57",
  "horizon": 5,
  "ts_iri": "https://www.theworldavatar.com/kg/pms_dh/Measure_e2bdc59b-3475-4962-8ebf-02f93996e144",
  "forecast_start_date": "Mon, 13 Jan 2020 00:00:00 GMT",
  "loaded_data_bounds": {
    "lowerbound": "2020-01-12T19:00:00Z",
    "upperbound": "2020-01-13T04:00:00Z"
  },
  "model_input_interval": [
    "Sun, 12 Jan 2020 19:00:00 GMT",
    "Sun, 12 Jan 2020 23:00:00 GMT"
  ],
  "model_output_interval": [
    "Mon, 13 Jan 2020 00:00:00 GMT",
    "Mon, 13 Jan 2020 04:00:00 GMT"
  ],
  "created_at": "Mon, 28 Nov 2022 02:55:58 GMT",
  "unit": "http://www.ontology-of-units-of-measure.org/resource/om-2/megawattHour",
  "time_format": "YYYY-MM-DDThh:mm:ssZ",
  "forecast_iri": "https://www.theworldavatar.com/kg/pms_dh/Forecast_66b47931-1438-4b00-9dec-90a28725d068",
  "status": "200"
}
```